### PR TITLE
Add option to skip sparse on pull requests

### PR DESCRIPTION
**Pull request by cgmb**
The CI will build and test without rocSPARSE when the ci:no-sparse label is added to a pull request.

### CODE REVIEW

File: .jenkins/common.groovy [hunk 1]
```diff
@@ -9,31 +9,39 @@ def runCompileCommand(platform, project, jobName, boolean sameOrg=false, boolean
     String hipClang = ''
     String debug = project.buildName.contains('Debug') ? '-g' : ''
     String centos = platform.jenkinsLabel.contains('centos') ? 'source scl_source enable devtoolset-7' : ''
-    String noOptimizations = ''
+    String[] options = []
```
Comment (jmachado-amd) [inline .jenkins/common.groovy]: Small typo here? I think you mean to use `List<String> options = []` (and similarly `List<String> getDeps = []` below); lists have the left shift operator `<<` overloaded to append new elements out of the box, while array sizes should be immutable.

File: .jenkins/common.groovy [hunk 2]
```diff
@@ -9,31 +9,39 @@ def runCompileCommand(platform, project, jobName, boolean sameOrg=false, boolean
...
+        getDeps << auxiliary.getLibrary('rocSPARSE', platform.jenkinsLabel, null, sameOrg)
+    }
+    getDeps << auxiliary.getLibrary('rocPRIM', platform.jenkinsLabel, null, sameOrg)
     def command = """#!/usr/bin/env bash
                 set -x
                 cd ${project.paths.project_build_prefix}
-                ${getRocBLAS}
-                ${getRocSPARSE}
-                ${getRocPRIM}
+                ${getDeps.join('\\n')}
```
Comment (jmachado-amd) [inline .jenkins/common.groovy:43]: This is just a heads-up in the case the builds keep failing. In my head `${getDeps.join('\\n')}` parses to something like `rocBLAS\nrocSPARSE\nrocPRIM`, and I don't see where the `\n`s are converted to new lines afterwards.

Comment (cgmb) [reply .jenkins/common.groovy:43]: It's nested within two strings. The `"""` string interprets the `\\` as `\` and the `'` interprets the `\n`.

**Review by jmachado-amd — APPROVED** [review]
Looks good to me!

**Review by tfalders — APPROVED** [review]

**Comment by cgmb** [comment]
I'll do a quick check without the label and then merge.

**Pull request merged**